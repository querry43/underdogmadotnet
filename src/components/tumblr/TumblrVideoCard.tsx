import * as React from 'react'
import ReactPlayer from 'react-player'

import TumblrCard from './TumblrCard'

const playerStyle : React.CSSProperties = {
  margin: 'auto'
}

class TumblrVideoCard extends TumblrCard {
  protected content() {
    return (
      <ReactPlayer style={playerStyle} url={this.props.post.permalink_url} />
    )
  }
}

export default TumblrVideoCard

// <iframe width="560" height="315" src="https://www.youtube.com/embed/DLxw6x9lTwM" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
