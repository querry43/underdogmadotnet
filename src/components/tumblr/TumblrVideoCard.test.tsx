/* tslint:disable:object-literal-sort-keys */

import * as React from 'react'

import TumblrVideoCard from './TumblrVideoCard'

it('matches snapshot', () => {
  const post = {
    "type": "video",
    "blog_name": "qrry43",
    "blog": {},
    "id": 179739805093,
    "post_url": "https://qrry43.tumblr.com/post/179739805093",
    "slug": "",
    "date": "2018-11-04 02:56:56 GMT",
    "timestamp": 1541300216,
    "state": "published",
    "format": "html",
    "reblog_key": "BAfm55p0",
    "tags": {},
    "short_url": "https://tmblr.co/ZlGD6Y2dPKyMb",
    "summary": "",
    "is_blocks_post_format": false,
    "recommended_source": null,
    "recommended_color": null,
    "note_count": 0,
    "caption": "",
    "reblog": {},
    "trail": {},
    "permalink_url": "https://www.youtube.com/watch?v=glZnkpIDWSE",
    "html5_capable": true,
    "video": {},
    "thumbnail_url": "https://i.ytimg.com/vi/glZnkpIDWSE/hqdefault.jpg",
    "thumbnail_width": 480,
    "thumbnail_height": 360,
    "player": {},
    "video_type": "youtube",
    "can_like": false,
    "can_reblog": false,
    "can_send_in_message": true,
    "can_reply": false,
    "display_avatar": true
  }

  expect(<TumblrVideoCard post={post} />).toMatchSnapshot()
})
