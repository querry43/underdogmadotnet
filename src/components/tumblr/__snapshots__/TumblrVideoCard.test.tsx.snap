// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<TumblrVideoCard
  post={
    Object {
      "blog": Object {},
      "blog_name": "qrry43",
      "can_like": false,
      "can_reblog": false,
      "can_reply": false,
      "can_send_in_message": true,
      "caption": "",
      "date": "2018-11-04 02:56:56 GMT",
      "display_avatar": true,
      "format": "html",
      "html5_capable": true,
      "id": 179739805093,
      "is_blocks_post_format": false,
      "note_count": 0,
      "permalink_url": "https://www.youtube.com/watch?v=glZnkpIDWSE",
      "player": Object {},
      "post_url": "https://qrry43.tumblr.com/post/179739805093",
      "reblog": Object {},
      "reblog_key": "BAfm55p0",
      "recommended_color": null,
      "recommended_source": null,
      "short_url": "https://tmblr.co/ZlGD6Y2dPKyMb",
      "slug": "",
      "state": "published",
      "summary": "",
      "tags": Object {},
      "thumbnail_height": 360,
      "thumbnail_url": "https://i.ytimg.com/vi/glZnkpIDWSE/hqdefault.jpg",
      "thumbnail_width": 480,
      "timestamp": 1541300216,
      "trail": Object {},
      "type": "video",
      "video": Object {},
      "video_type": "youtube",
    }
  }
/>
`;
